subsys/bluetooth/controller/ll_sw/nordic/lll/lll_sync_iso.c

```diff
@@ -691,18 +699,31 @@ static void isr_rx(void *param)
 	/* Find the next repetition (irc_curr)th subevent to receive PDU */
 	if (lll->irc_curr < lll->irc) {
 		if (!new_burst) {
-			uint8_t payload_index;
+			uint32_t payload_offset;
+			uint16_t payload_index;
 
 			/* Increment to next repetition count and be at first
 			 * burst count for it.
 			 */
 			lll->bn_curr = 1U;
 			lll->irc_curr++;
 
+			/* Check payload buffer overflow */
+			/* FIXME: Typically we should not have high latency, but do have an
+			 *        assertion check to ensure we do not rollover in the payload_index
+			 *        variable use. Alternatively, add implementation to correctly
+			 *        skip subevents as buffers at these high offset are unavailable.
+			 */
```

Comment:
I assume there is some correlation between offset and latency here?

Reply:
Here latency is the skipped ISO intervals, and offset is the PDU buffer array offset for a particular buffer window of reception.

The reception is implemented as a sliding window of PDU buffer pointers; each window being able to receive the current payload counter associated PDUs and any subsequent pre-transmission PDUs.
But when ISO intervals are skipped, then there will be many PDU buffer windows that have not generated `invalid` status PDUs to ISOAL layer and hence for the current payload counter the PDU buffer index is well ahead into the buffer array.